.github/workflows/main.yml

```
@@ -173,6 +174,37 @@ jobs:
             OS-${{ runner.os }},
             VM-${{ matrix.platform }},
             Py-${{ steps.python-install.outputs.python-version }}
+            
+  check-changed-folders:
+    name: Check files
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Get changed files in the _distutils folder
+        id: changed-files-specific-distutils
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_distutils/**
+      - name: Get changed files in the _vendor folder
+        id: changed-files-specific-vendor
+        uses: tj-actions/changed-files@v34
+        with:
+          files: |
+            setuptools/_vendor/**
+      - name: Run step if any file(s) in the _distutils folder change
+        if: steps.changed-files-specific-distutils.outputs.any_changed == 'true'
+        run: |
+          echo "One or more files in the setuptools/_distutils folder has changed." >> $GITHUB_STEP_SUMMARY
```

I still think that piping to `tee` would be better:
```suggestion
          echo "One or more files in the setuptools/_distutils folder has changed." | tee "${GITHUB_STEP_SUMMARY}"
```